### DISCUSSION TITLE
Missing TargetFramework on Publish

### DISCUSSION THREAD

**Discussion by mhrstv** [unoplatform/uno#18718] [Q&A] [answered]
Hello,
I am trying to publish my project for the <ins>net8.0-desktop</ins> framework and unfortunately get an error for a missing _**TargetFramework**_ within my '<ins>project.assets.json</ins>'.

The framework is not the desktop one, but <ins>net8.0-maccatalyst</ins>.
Seems like it is going through the other frameworks as well, even though I'm not using them in the configuration profile.
##
**Here's the whole error**:
<sub>Error (active)	NETSDK1047	Assets file 'E:\kodify\Kodify\obj\project.assets.json' doesn't have a target for 'net8.0-maccatalyst/maccatalyst-arm64'. Ensure that restore has run and that you have included 'net8.0-maccatalyst' in the TargetFrameworks for your project. You may also need to include 'maccatalyst-arm64' in your project's RuntimeIdentifiers.	Kodify (net8.0-maccatalyst)	C:\Program Files\dotnet\sdk\8.0.403\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets	266	</sub>
##

<ins>Publish profile settings</ins>:
**Configuration:** _Release_
**Target Framework:** _net8.0-desktop_
**Deployment mode:** _Framework-dependant_
**Target runtime:** _Portable_
**Target location:** _(bin\Release\net8.0-desktop\publish\)_

**Comment by jeromelaban** [comment]
Thanks for the report. Can you [provide a binlog](https://platform.uno/docs/articles/uno-builds-troubleshooting.html) of the build?

**Reply by mhrstv** [reply]
[Kodify_Debug_AnyCPU_Build_2024-11-06T14_37_31.5627605+02_00.zip](https://github.com/user-attachments/files/17647210/Kodify_Debug_AnyCPU_Build_2024-11-06T14_37_31.5627605%2B02_00.zip)

**Reply by mhrstv** [reply]
It's not even the error that I mentioned earlier.
The one in the binlog happens regularly with my favicon.ico on browserwasm.
Fixes after I delete and run the project a few times, sometimes.
Can't seem to understand why it happens and honestly need to fix it.

**Reply by mhrstv** [reply]
Managed to get rid of the **_maccatalyst_** error, but can't seem to fix the other one.
The one in the binlog I mean.

**Reply by jeromelaban** [reply]
Thanks. Can you re-create the binlog with the diagnostics log level?

**Reply by mhrstv** [reply]
I will. For now the error seems to have stopped occuring. It doesn't happen all the time.

**Reply by jeromelaban** [reply, marked as answer]
Thanks, this helps, we'll take a look.

**Reply by mhrstv** [reply]
Glad my "issue" has provided decent feedback. Although it seems to resolve on its own, I believe It'd be great for it to not occur at all.
If I happen to stumble upon anything similar again I'll open a discussion again.
Thanks for your help.

**Reply by jeromelaban** [reply]
One more thing, I notice that you're using an older version of the Uno.Sdk. Does it still happen with the latest?

**Reply by mhrstv** [reply]
I haven't come to upgrading to the latest yet. Should I? I know it has many improvements, bug fixes & features, though I'm not sure whether It'll become problematic for a reason or another.

**Reply by jeromelaban** [reply]
The best way to know is to try :)
